From linux-usb-devel-bounces@lists.sourceforge.net  Sun Sep  9 09:51:52 2007
From: Lamarque Vieira Souza <lamarque@syst.com.br>
Date: Tue, 4 Sep 2007 12:15:08 -0300
Subject: USB: More USB_QUIRK_RESET_RESUME devices
To: linux-usb-devel@lists.sourceforge.net
Message-ID: <200709041215.08535.lamarque@syst.com.br>


I would like have the attached patch added to Linux kernel. The three
usb flash memories listed in the patch are being used in Intel's
ClassmatePC and need USB_QUIRK_RESET_RESUME to work reliably when
resuming from ram.

---
 drivers/usb/core/quirks.c |    9 +++++++++
 1 file changed, 9 insertions(+)

--- a/drivers/usb/core/quirks.c
+++ b/drivers/usb/core/quirks.c
@@ -76,6 +76,12 @@ static const struct usb_device_id usb_qu
 	/* Genesys USB-to-IDE */
 	{ USB_DEVICE(0x0503, 0x0702), .driver_info = USB_QUIRK_NO_AUTOSUSPEND },
 
+	/* INTEL VALUE SSD */
+	{ USB_DEVICE(0x8086, 0xf1a5), .driver_info = USB_QUIRK_RESET_RESUME },
+
+	/* M-Systems Flash Disk Pioneers */
+	{ USB_DEVICE(0x08ec, 0x1000), .driver_info = USB_QUIRK_RESET_RESUME },
+
 	/* Agfa Snapscan1212u */
 	{ USB_DEVICE(0x06bd, 0x2061), .driver_info = USB_QUIRK_NO_AUTOSUSPEND },
 	/* Seagate RSS LLC */
@@ -100,6 +106,9 @@ static const struct usb_device_id usb_qu
 	/* Apple iPhone */
 	{ USB_DEVICE(0x05ac, 0x1290), .driver_info = USB_QUIRK_NO_AUTOSUSPEND },
 
+	/* SKYMEDI USB_DRIVE */
+	{ USB_DEVICE(0x1516, 0x8628), .driver_info = USB_QUIRK_RESET_RESUME },
+
 	{ }  /* terminating entry must be last */
 };
 
